clients/ctl/admin-ui/src/features/taxonomy/hooks.tsx

```diff
@@ -104,15 +111,24 @@ export const useDataUse = (): TaxonomyHookData<DataUse> => {
   };
 
   const [edit] = useUpdateDataUseMutation();
+  const [create] = useCreateDataUseMutation();
+
+  const transformFormValuesToEntity = (formValues: DataUse) => ({
+    ...formValues,
+    // text inputs don't like having undefined, so we originally converted
+    // to "", but on submission we need to convert back to undefined
```

Comment:
Seems like this could come up for a variety of fields. The problem of `undefined` as an initial value is probably that the underlying React input element thinks it should be uncontrolled. `CustomTextInput` could look at the Formik `field` value and prevent that problem.  Though that would only make line 142 unnecessary -- this function would still need to exist, as transforming the values on the field [isn't supported yet.](https://github.com/jaredpalmer/formik/issues/2488#issuecomment-814244781)

Reply:
oh neat, thanks for the link! yeah, I find myself doing this all the time to avoid the uncontrolled warning 😕 I think I'll hold off on putting logic in `CustomTextInput` right now, since I'm worried it'll be confusing if we are passing `undefined` into it, yet the value we get back is always a string